src/core/internal/cpptrace.d

```diff
@@ -0,0 +1,119 @@
+/**
+* This module provides the demangling of C++ symbols for stacktrace
+*
+* Copyright: Copyright © 2020, The D Language Foundation
+* License: $(HTTP www.boost.org/LICENSE_1_0.txt, Boost License 1.0).
+* Authors: Ernesto Castellotti
+* Source: $(DRUNTIMESRC core/internal/_cppdemangle.d)
+*/
+module core.internal.cpptrace;
+
+import core.internal.cppdemangle : CPPDemangle, CPPDemangleStatus;
+import core.stdc.stdio : fprintf, stderr;
+import core.stdc.stdlib : exit;
+
+/**
+* Demangles C++ mangled names passing the runtime options to cppdemangle.
+*
+* If it is not a C++ mangled name or cppdemangle is not supported by your platform, the original mangled C++ name will be returned.
+* The optional destination buffer and return will be contains the same string if the demangle is successful.
+* This function is used to demangle C++ symbols in the stacktrace.
+*
+ * Params:
+ *  buf = The string to demangle.
+ *  dst = The destination buffer, if the size of the destination buffer is <= the size of cppdemangle output the return string would be incomplete.
+ *
+ * Returns:
+ *  The demangled name or the original string if the name is not a mangled C++ name.
+ */
+char[] demangleCppTrace(const(char)[] buf, char[] dst)
+{
+    import core.internal.cppdemangle : copyResult;
+
+    if (!CPPTrace.config.enable)
+    {
+        return copyResult(buf, dst);
+    }
+
+    return CPPTrace.instance.cppdemangle(buf, dst, !CPPTrace.config.noprefix, CPPTrace.config.prefix);
+}
+
+private struct CPPTrace
+{
+    __gshared CPPDemangle instance;
+    __gshared Config config;
+
+    static this()
```

Comment:
A `static this()` is run on each thread constructions. You want a `shared static this` here, no ?